src/external/wannier90.jl

```diff
@@ -287,3 +289,79 @@ default_wannier_centres(n_wannier) = [rand(1, 3) for _ in 1:n_wannier]
     @timing "Wannierization" wannier90_jll.wannier90(exe -> run(Cmd(`$exe $prefix`; dir)))
     fileprefix
 end
+
+@doc raw"""
+    For a given set of bands ``\{Ψ_{nk}\}_{1\leq n \leq N_in}``
+    and a Wannier90 file containing unitary matrices (for disentanglement or
+    Wannierization) returns a set of band ``{Ψ_out_{nk}}_{1\leq n \leq N_out}``
+    with:
+    ``Ψ_out_{nk} = \sum\limits_{m=1}^{N_in} U_{mn}Ψ_{nk}``
+    ``N_in > N_out`` for disentanglement.
+    ``N_in = N_out`` for wannierization.
+"""
+function apply_U_matrices(file, ψ)
+    # Extract all Uk
+    @assert(isfile(file))
+    Uks = readdlm(file); num_kpts, num_ψout, num_ψin = Uks[2,1:3]
+    len_Uk = num_ψin*num_ψout
+
+    ψ_out = copy(ψ) # Better ideas than copy to inherit the type of ψ in ψ_out ?
+    for k in 1:num_kpts
+        # Locate the beginning of the matrix Uk written as a vector
+        i_Uk = 4 + (k-1)*(len_Uk+1)
+        Uk = reshape([x+im*y for (x,y) in eachrow(Uks[i_Uk:i_Uk+(len_Uk-1), 1:2])],
+                     num_ψin, num_ψout)
+        # Check that Uk is unitary
```

Comment:
unitary only applies to square matrices. People usually write "has orthogonal columns"

Reply:
also comment is redundant with the following